src/Domain/Creators/BinaryCompactObject.hpp

```diff
@@ -363,13 +351,19 @@ class BinaryCompactObject : public DomainCreator<3> {
   };
 
   struct InitialRefinement {
-    using type = size_t;
+    using type =
+        std::variant<size_t, std::array<size_t, 3>,
+                     std::vector<std::array<size_t, 3>>,
+                     std::unordered_map<std::string, std::array<size_t, 3>>>;
     static constexpr Options::String help = {
         "Initial refinement level. Applied to each dimension."};
```

Comment:
maybe expand the help string to explain what the different type options mean or point back to the main help string?